### ISSUE TITLE
UGE configured to dynamically allocate a GID when submitting a job

### ISSUE THREAD

**Issue by sylabskeith-zz** [apptainer/singularity#1303]
## Version of Singularity
2.4.2

### Expected behavior

Proper allocation of dynamic GID during job submission to UGE

###Actual Behavior

We are getting ENOENT errno on the following line of code:
https://github.com/singularityware/singularity/blob/2.4.2/src/lib/runtime/files/group/group.c#L147

### Steps to reproduce behavior

Some reference to UGE:
http://gridengine.eu/grid-engine-internals

**Comment by markeganfuller** [comment]
Running singularity on our local cluster which uses UGE just results in the following for the UGE job group:

```
VERBOSE [U=2361,P=12042]   singularity_runtime_files_group()         Skipping GID 20105 as group entry does not exist.
```

**Comment by dtrudg** [comment]
Looks like ENOENT needs to be handled as a possible 'group not found' see:

`man 3 getgrgid`

> 0 or ENOENT or ESRCH or EBADF or EPERM or ...
The given name or gid was not found.

**Comment by dtrudg** [comment]
@sylabskeith - is it possible to test PR #1366? This will handle an ENOENT on the group query, and result in same behaviour that .@markeganfuller was seeing above on his system.

**Comment by GodloveD** [comment]
^ Also @paulo7777

**Comment by dtrudg** [comment]
Closing this as it is understood fixed with #1366 which has been merged. Please don't hesitate to reopen if any further issue.
